notebooks/12-xarray-advanced.md

```diff
@@ -0,0 +1,686 @@
+ghent_precipitation_month_max = ghent_precipitation.groupby("time.month").max()*1000
+
+fig, ax = plt.subplots()
+ghent_precipitation_month_max.to_dataframe()["precipitation_m"].plot.barh(ax=ax)
+ax.set_xlabel("Maximal monthly precipitation (mm)")
+ax.set_ylabel("Month of the year")
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+    
+Calculate the pixel-based maximal temperature _for each season_. Make a plot (`imshow`) with each of the seasons in a separate subplot next to each other. 
+
+<details><summary>Hints</summary>
+    
+* You need to group based on the season of the data point, hence `groupby("...")` with `time` as an existing dimension, the `time.season` shortcut to get the corresponding season will work.
+* The labels of the season groups are sorted on the strings in the description instead of the seasons. The workaround is to `sortby` and provide a correctly sorted version to sort the `season` with, see the open issue https://github.com/pydata/xarray/issues/757.
+* Use facetting to plot each of the seasons next to each other, with `col="season"`.
+
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+seaons_temp = era5_renamed["temperature_k"].groupby("time.season").max()
+# See https://github.com/pydata/xarray/issues/757 for getting well-sorted groups for plotting
+seaons_temp = seaons_temp.sortby(xr.DataArray(['DJF','MAM','JJA', 'SON'],dims=['season']))
+seaons_temp.plot.imshow(col="season", cmap="Reds")
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+    
+Create the yearly total snowfall from 1991 up to 2005 and convert the snowfall into cm.  Make a plot (`imshow`) with each of the individual years in a separate subplot divided into 3 rows and 5 columns.
+    
+Make sure to update the name of the snowfall variable and/or colorbar label to make sure it defines the unit in cm.
+ 
+
+<details><summary>Hints</summary>
+
+* When selecting time series data from a coordinate with datetime-aware data, one can use strings to define a date. In combination with `slice`, the selection of the required years becomes `slice("1991", "2005")`.
+* From montly to yearly data is a `resample` of the data.
+* Use `.rename(NEW_NAME)` to update the name of a `DataArray`
+* xarray supports _facetting_ directly, check out the `col` and `col_wrap` parameters in the plot functions of xarray or check http://xarray.pydata.org/en/stable/user-guide/plotting.html#faceting.
+* To update the colorbar unit use, the `cbar_kwargs` option.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+snowfall_1991_2005 = era5_renamed.sel(time=slice("1991", "2005"))["snowfall_m"]
+snowfall_yearly = snowfall_1991_2005.resample(time="Y").sum()*100
+snowfall_yearly = snowfall_yearly.rename("snowfall_cm")
+snowfall_yearly.plot.imshow(col="time", col_wrap=5, cmap=cmocean.cm.ice, 
+                            cbar_kwargs={"label": "snowfall (cm)"})
+```
+
```

Comment:
Another idea for an exercise: sum over longitude/latitude to get eg average temperature for full region, and then make a timeseries plot. The sum over lon/lat is included once in the previous set of exercises, but might be good to repeat once here.

(not that we don't have enough exercises of course ;))

Reply:
This is maybe a bit too similar to the one in the previous set?
![image](https://user-images.githubusercontent.com/754862/135731903-b16c1abd-ac9d-48f2-98cc-85f4a321f20c.png)

It is a time series plot of the snowfall.... Maybe in combination with a groupby/rolling?

Reply:
ok, I added an exercise combining the sel over two dimensions with a resample operation for temperature.